core/src/main/java/org/elasticsearch/action/support/AbstractListenableActionFuture.java

```diff
@@ -50,7 +50,8 @@ public void addListener(final ActionListener<T> listener) {
     }
 
     public void internalAddListener(ActionListener<T> listener) {
-        listener = new ThreadedActionListener<>(logger, threadPool, ThreadPool.Names.LISTENER, listener, false);
+        listener = new ThreadedActionListener<>(logger, threadPool, ThreadPool.Names.LISTENER,
```

Comment:
hmm.. is this correct? It's debatable what we should do here. Should we restore the context in which the listener was added or should we use the context of the thread that made the request and also keep the response headers. The way this is set up now, consumers don't have a choice. I'm starting to doubt if we should do this... @s1monw wdyt?